backend/auth/role_map_repository.go

```diff
@@ -393,36 +127,97 @@ func GetRoleMapConfig(namespace string, name string) (map[string]*models.Role, m
 
 	details := (*res.ResourceDetails).(*unstructured.Unstructured)
 
-	roleMapData, foundRoleMap, err2 := unstructured.NestedString(details.Object, "data", "role-map")
-	if err2 != nil || !foundRoleMap {
+	roleMapConfigData, foundRoleMapConfig, err2 := unstructured.NestedString(details.Object, "data", "role-map")
+	if err2 != nil || !foundRoleMapConfig {
 		log.Printf("Error retrieving roleMap data: %v", err)
 		return nil, nil
 	}
 
-	roleMap := make(map[string]*models.Role)
-	err2 = yaml.Unmarshal([]byte(roleMapData), &roleMap)
+	roleMapConfig := make(map[string]*roleConfig)
+	err2 = yaml.Unmarshal([]byte(roleMapConfigData), &roleMapConfig)
 	if err2 != nil {
 		log.Printf("Error parsing roleMap data: %v", err)
 		return nil, nil
 	}
 
-	subroleMapData, foundSubRoleMap, err2 := unstructured.NestedString(details.Object, "data", "subrole-map")
-	subroleMap := make(map[string]*models.Role)
+	subroleMapConfigData, foundSubroleMapConfig, err2 := unstructured.NestedString(details.Object, "data", "subrole-map")
+	subroleMapConfig := make(map[string]*roleConfig)
+
+	var subroleMap map[string]*models.Role
 
-	if !foundSubRoleMap || err2 != nil {
+	if !foundSubroleMapConfig || err2 != nil {
 		log.Printf("Error retrieving subroleMap data")
 	} else {
-		err2 = yaml.Unmarshal([]byte(subroleMapData), &subroleMap)
+		err2 = yaml.Unmarshal([]byte(subroleMapConfigData), &subroleMapConfig)
 		if err2 != nil {
 			log.Printf("Error parsing subroleMap data: %v", err)
 			// No return as subroleMap is optional roleMap can be used without it
 		}
-
+		subroleMap = fromRoleMapConfig(subroleMapConfig)
 		if hasCycle(subroleMap) {
 			log.Printf("Cycle detected in subrole map")
 			subroleMap = make(map[string]*models.Role) // clear subrole map, can't use it
 		}
 	}
 
-	return roleMap, subroleMap
+	return fromRoleMapConfig(roleMapConfig), subroleMap
+}
+
+func fromRoleMapConfig(config map[string]*roleConfig) map[string]*models.Role {
+	roleMap := make(map[string]*models.Role)
+	for name, roleConfig := range config {
+		role := fromRoleConfig(roleConfig)
+		if role.Name == "" {
+			role.Name = name
+		}
+		roleMap[name] = role
+	}
+	return roleMap
+}
+
+func fromRoleConfig(config *roleConfig) *models.Role {
+	permit := fromOperationConfigList(config.Permit)
+	deny := fromOperationConfigList(config.Deny)
+
+	role := &models.Role{
+		Name:     config.Name,
+		Subroles: config.Subroles,
+	}
+	if len(permit) > 0 {
+		role.Permit = permit
+	}
+	if len(deny) > 0 {
+		role.Deny = deny
+	}
+	return role
+}
```

Comment:
Does if len(permit) > 0 actually do anything? If permit is empty, wouldn’t it still just default to being empty anyway?

```suggestion
func fromRoleConfig(config *roleConfig) *models.Role {
    if config == nil {
        return &models.Role{}
    }
    return &models.Role{
        Name:     config.Name,
        Permit:   fromOperationConfigList(config.Permit),
        Deny:     fromOperationConfigList(config.Deny),
        Subroles: config.Subroles,
    }
}
```

Reply:
It matters, the difference is nil value vs empty list